src/main/java/com/einabit/client/EinabitClient.java

```diff
@@ -84,7 +93,7 @@ public void tap(final String variable, final EinabitServerListener callback) {
             int readBytes;
 
             byte[] buffer = new byte[BUFFER_SIZE];
-            while ((readBytes = dataInputStream.read(buffer)) != -1) {
+            while ((readBytes = dataInputStream.read(buffer)) != -1 && !Thread.currentThread().isInterrupted()) {
```

Comment:
Can you please take a look on this?

https://github.com/Einabit/client.java/pull/10/files#diff-2596d8b447cd46c7d81efc77d9439d94c2fda5515c77db849bbe0a959923676eR81-R94

We need to run the while loop until the thread is interrupted, the server or the datainputstream shouldn't drive client behavior. The datainputstream may be empty for a period of time if the server does not find any changes to inform.

thoughts?